doc/DHCPv6_relay/DHCPv6-relay-agent-High-Level-Design.md

```diff
@@ -0,0 +1,215 @@
+
+options: include a &quot;Relay Message option&quot;
+
+# DHCPv6 Packet Forwarding
+
+The DHCPv6 relay agent on the routing switch forwards DHCPv6 client packets to all DHCPv6 servers that are configured in the table administrated for each VLAN.
+
+A DHCPv6 client locates a DHCPv6 server using a reserved, link-scoped multicast address.
+
+The packets are forwarded using multicasting to a group of DHCPv6 servers
+
+# Relay Agent Behavior
+
+1. DHCPv6 client sends multicast SOLICITE message to ALL\_DHCP\_Relay\_Agents\_and\_Servers. Message received by relay agent.
+  - Relay agent at default uses ALL\_DHCP\_Servers multicast address. It may be configured to use unicast addresses, or other addresses selected by the network administrator.
+2. DHCPv6 relay agent constructs a Relay-forward message copies the source address from header of the IP datagram to the peer-address field of the Relay-forward message and received DHCP message into Relay Message option, and relays this Relay-forward message to the DHCPv6 server in RELAY\_FORWARD message
+  - DHCPv6 relay agent also places a global or site-scope address with a prefix assigned to the link on which the client should be assigned an address in the link-address field. (will be used by server to determine the link from which the client should be assigned an address)
+  - Hop-count in Relay-forward message is set to 0.
+  - If Relay Agent were to relay a message from a relay agent, it checks if the hop-count in the message is greater than or equal to HOP\_COUNT\_LIMIT, and discard if so. Else, hop\_count is incremented by 1.
+3. DHCPv6 server received the SOLICIT message, refers to the Relay Agent IP and select an IP address to allocate to the DHCPv6 client.
+4. The DHCPv6 server constructs a RELAY-REPLY message that embeds the advertise messages, and sends it to the DHCPv6 relay agent.
+5. DHCPv6 client receives ADVERTISE message and relays a REQUEST message to the DHCPv6 relay agent.
+6. DHCPv6 relay agent constructs REQUEST message into a RELAY-FORWARD message, and relays to DHCPv6 server.
+7. DHCPv6 server receives the REQUEST message, and sends a REPLY message to the relay agent. Server creates a Relay-reply message that includes a Relay Message option containing the the REPLY message and sends it to the relay agent.
+8. DHCPv6 relay agent extracts message and relays the message to the address contained in the peer-address field of the Relay-reply message.
+9. DHCPv6 client receives the REPLY message that contains the desired IP address.
+
+![image](https://user-images.githubusercontent.com/42761586/117859842-5fd13b00-b244-11eb-9297-c2674d128dd9.png)
+
+# Requirements
+
+- Configured and running DHCPv6 client and server
+- Connectivity between the relay agent and DHCPv6 server
+- Configure one or more IP helper addresses for specified VLANs to forward DHCPv6 requests to DHCPv6 servers on other subnets.
+- IPv6 multicast address for all DHCP agents: IPv6 ff02::1:2
```

Comment:
nit: ALL DHCPv6 Servers and agents multicast address.

Reply:
Why is this a requirement? please provide details.

Reply:
This is not a requirement. I mistakenly thought multicasting was used for our relay agent. Made the revision!